pkg/asset/installconfig/gcp/client.go

```diff
@@ -80,6 +81,40 @@ func (c *Client) GetMachineType(ctx context.Context, project, zone, machineType
 	return req, nil
 }
 
+// GetMachineTypeWithZones retrieves the specified machine type and the zones in which it is available.
+func (c *Client) GetMachineTypeWithZones(ctx context.Context, project, region, machineType string) (*compute.MachineType, sets.Set[string], error) {
+	svc, err := c.getComputeService(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	var machine *compute.MachineType
+	zones := sets.New[string]()
+	filter := fmt.Sprintf("name = \"%s\" AND zone : %s-*", machineType, region)
+	req := svc.MachineTypes.AggregatedList(project).Filter(filter).Context(ctx)
+	err = req.Pages(ctx, func(page *compute.MachineTypeAggregatedList) error {
+		for _, scopedList := range page.Items {
+			for _, item := range scopedList.MachineTypes {
+				if machine == nil {
+					machine = item
```

Comment:
maybe if we do what i  suggested in the previous comment, this can also be referenced to the first one?